core/src/main/java/org/apache/calcite/rel/type/RelDataTypeFactoryImpl.java

```diff
@@ -603,6 +603,32 @@ public RelDataType getComponentType() {
       }
     }
 
+    /**
+     * For {@link JavaType} created with {@link Map} class,
+     * cannot get the type of key and value.
+     * Using ANY type as key type.
+     */
+    @Override public RelDataType getKeyType() {
+      if (Map.class.isAssignableFrom(clazz)) {
+        return createSqlType(SqlTypeName.ANY);
```

Comment:
Say the map is by <Integer, Integer>
Why return SqlTypeName.ANY here ?

Reply:
Because I cannot find a way to get the `Integer` parameter type when create type with Map class.
And use `SqlTypeName.ANY` can cover other types.